themes/ddbasic/scripts/misc.js

```diff
@@ -34,6 +34,11 @@
       close_messages_button.on('click', function(){
         pane_messages.slideUp('fast');
       });
+
+      // Move ting subsearch message to the top of the page.
+      $('.ting-subsearch-message', context).once('ddbasic-ting-subsearch-message', function() {
+        $('.pane-page-content').prepend($(this).parent());
+      });
```

Comment:
I have considered making a ctools content_type plugin and place it in top of main content on ting search panel page. This way we would have easy access to the user search via contexts, and we can be assured that we are [rendered after the search result](https://github.com/ding2/ding2/blob/master/modules/ting_search/plugins/content_types/search_result.inc#L28), so we have access to the current search results.
But the main content in the panel layout is not full width either, so we will still have to place it in another container in frontend to get it full width in top.

Reply:
If you have a suggestion on how to implement this a cleaner way I'll be happy to try to implement it.

I'm all out of ideas.

Reply:
Added top banner to our panel page layout. Added new ctools content_types plugins for suggestions and translate and placed them in new top banner.

Subsearch bibliotek.dk is still using `hook_ting_search_results_prefix()` and `hook_ting_search_results_suffix()` as that was the easiest because of the position setting.

Suggestions still performs its zero hit redirect in `hook_ting_search_results_prefix()`, since that hook has the best timing right after ting_search is executed. Also now using `drupal_set_message()` instead to display info to user about redirect.